src/common/utxobased/plugin/makeUtxoEngine.ts

```diff
@@ -271,6 +274,10 @@ export async function makeUtxoEngine(config: EngineConfig): Promise<EdgeCurrency
         privateKeys
       })
 
+      for (const address of transaction.ourReceiveAddresses) {
```

Comment:
You shouldn't mark the address "used" at this stage. It should happen on "saveTx" so when that is implemented move it there.